string closest_node
string current_node
string current_edge
